schema/core.go

```diff
@@ -175,32 +175,52 @@ func (a CoreAttribute) validate(attribute interface{}) (interface{}, *errors.Sci
 		return nil, nil
 	}
 
-	if a.multiValued {
-		// return false if the multivalued attribute is not a slice.
-		arr, ok := attribute.([]interface{})
-		if !ok {
-			return nil, &errors.ScimErrorInvalidSyntax
+	if !a.multiValued {
+		return a.validateSingular(attribute)
+	}
+
+	switch arr := attribute.(type) {
+	case map[string]interface{}:
+		// return false if the multivalued attribute is empty.
+		if a.required && len(arr) == 0 {
+			return nil, &errors.ScimErrorInvalidValue
+		}
+
+		for _, subArr := range a.subAttributes {
+			if v, ok := arr[subArr.name]; ok {
+				fmt.Println(v)
+				fmt.Printf("v.Type=%T\n", v)
+				fmt.Printf("subArr.typ=%+v\n", subArr.typ)
```

Comment:
I think you forgot to remove these?

Reply:
Oh, thanks.
Fixed it. https://github.com/elimity-com/scim/pull/106/commits/4640d9c9d1ee29dec857492a99acac71a6c380e7